src/metalperformanceshadersgraph.cs

```diff
@@ -0,0 +1,1831 @@
+		[Export ("softMaxCrossEntropyGradientWithIncomingGradientTensor:sourceTensor:labelsTensor:axis:reductionType:name:")]
+		MPSGraphTensor SoftMaxCrossEntropyGradient (MPSGraphTensor gradientTensor, MPSGraphTensor sourceTensor, MPSGraphTensor labelsTensor, nint axis, MPSGraphLossReductionType reductionType, [NullAllowed] string name);
+	}
+
+	// @interface MPSGraphMatrixMultiplicationOps (MPSGraph)
+	[iOS (14,0), TV (14,0), Mac (11,0), MacCatalyst (14,0)]
+	[Category]
+	[BaseType (typeof(MPSGraph))]
+	interface MPSGraph_MPSGraphMatrixMultiplicationOps
+	{
+		// -(MPSGraphTensor * _Nonnull)matrixMultiplicationWithPrimaryTensor:(MPSGraphTensor * _Nonnull)primaryTensor secondaryTensor:(MPSGraphTensor * _Nonnull)secondaryTensor name:(NSString * _Nullable)name __attribute__((swift_name("matrixMultiplication(primary:secondary:name:)")));
+		[Export ("matrixMultiplicationWithPrimaryTensor:secondaryTensor:name:")]
+		MPSGraphTensor MatrixMultiplication (MPSGraphTensor primaryTensor, MPSGraphTensor secondaryTensor, [NullAllowed] string name);
+	}
+
+	// @interface MPSGraphCreateSparseOpDescriptor : NSObject <NSCopying>
+	[TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject))]
+	interface MPSGraphCreateSparseOpDescriptor : NSCopying
+	{
+		// @property (readwrite, nonatomic) MPSGraphSparseStorageType sparseStorageType;
+		[Export ("sparseStorageType", ArgumentSemantic.Assign)]
+		MPSGraphSparseStorageType SparseStorageType { get; set; }
+
+		// @property (readwrite, nonatomic) MPSDataType dataType;
+		[Export ("dataType", ArgumentSemantic.Assign)]
+		MPSDataType DataType { get; set; }
+
+		// +(instancetype _Nullable)descriptorWithStorageType:(MPSGraphSparseStorageType)sparseStorageType dataType:(MPSDataType)dataType __attribute__((swift_name("sparseDescriptor(descriptorWithStorageType:dataType:)")));
+		[Static]
+		[Export ("descriptorWithStorageType:dataType:")]
+		[return: NullAllowed]
+		MPSGraphCreateSparseOpDescriptor Create (MPSGraphSparseStorageType sparseStorageType, MPSDataType dataType);
+	}
+
+	// @interface MPSGraphSparseOps (MPSGraph)
+	[Category]
+	[BaseType (typeof(MPSGraph))]
+	interface MPSGraph_MPSGraphSparseOps
+	{
+		// -(MPSGraphTensor * _Nonnull)sparseTensorWithType:(MPSGraphSparseStorageType)sparseStorageType tensors:(NSArray<MPSGraphTensor *> * _Nonnull)inputTensorArray shape:(MPSShape * _Nonnull)shape dataType:(MPSDataType)dataType name:(NSString * _Nullable)name __attribute__((availability(macos, introduced=12.0))) __attribute__((availability(ios, introduced=15.0))) __attribute__((availability(tvos, introduced=15.0))) __attribute__((swift_name("sparseTensor(sparseTensorWithType:tensors:shape:dataType:name:)")));
+		[TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
```

Comment:
You can put these attributes on the type, that way they take effect for all the members.

Also, that way we won't automatically add attributes the category saying it's available in the earliest OS version we support (since the type doesn't have attributes right now).